core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -97,7 +99,9 @@ public LocalDispatchQueryFactory(
         this.warningCollectorFactory = requireNonNull(warningCollectorFactory, "warningCollectorFactory is null");
         this.clusterSizeMonitor = requireNonNull(clusterSizeMonitor, "clusterSizeMonitor is null");
         this.executor = dispatchExecutor.getExecutor();
-        this.maxStateMachineThreadsPerQuery = requireNonNull(queryManagerConfig, "queryManagerConfig is null").getMaxStateMachineCallbackThreads();
+        requireNonNull(queryManagerConfig, "queryManagerConfig is null");
```

Comment:
This `requireNonNull` is unnecessary as queryManagerConfig is accessed on the next line. If it is a null we will get a similarly good error message from the next line (nullptr exception messages are improved by default from jdk 15 onwards)